nucypher/policy/conditions/evm.py

```diff
@@ -217,24 +242,23 @@ def _configure_provider(self, *args, **kwargs):
 
     def _get_unbound_contract_function(self) -> ContractFunction:
         """Gets an unbound contract function to evaluate for this condition"""
-        contract = self.w3.eth.contract(address=self.contract_address, abi=self.function_abi)
-        contract_function = getattr(contract.functions, self.method)  # TODO: Use function selector instead/also?
-        return contract_function
-
-    def _evaluate(self, **context) -> Tuple[bool, Any]:
-        """Performs onchain read and return value test"""
-        parameters, return_value_test = _resolve_any_context_variables(
-            self.parameters, self.return_value_test, **context
-        )
+        try:
+            contract = self.w3.eth.contract(
+                address=self.contract_address, abi=self.function_abi
+            )
+            contract_function = getattr(
+                contract.functions, self.method
+            )  # TODO: Use function selector instead/also?
+            return contract_function
+        except Exception as e:
+            raise ReencryptionCondition.InvalidCondition(
+                f"Unable to obtain contract function, '{self.method}', for condition: {e}"
```

Comment:
This line is suggests that including a "name" or "nickname" metadata field in the condition might be useful.